### PR TITLE
feat: add carplay and android auto support

### LINKED ISSUE #177: Apple CarPlay support
Can we use apple CarPlay with this SDK?

### PR DESCRIPTION
**Pull request by illuminati1911**
Adds support for Apple CarPlay and Android Auto.

- Added BaseAutoSceneDelegate for iOS and AndroidAutoBaseScreen for Android on a plugin level to manage setting up the base functionality for each platform.
- Added example of setting up each platform in the SampleApp. iOS SampleApp has a new separate target for CarPlay support `SampleAppCarPlay`.
- Added separate native module `NavAutoModule` to manage communication between RN code and the map instance.
- Added a generic event channel so that user actions can be sent from native code to RN. e.g. user presses a button in CarPlay window and this event is sent to RN code.
- Added a function to NavAutoModule to check whether CarPlay or Android Auto screen has been initialized and is available.

Resolves #177 

- [x] Tests pass
- [x] Appropriate changes to documentation are included in the PR

### CODE REVIEW

**Comment by ArturoSalazarB16** [comment]
Hi! Thanks a lot for working on this Ville/Joonas. Moving forward, is it possible to create unitary PRs to make reviewing easier? and would also be easier to debug/investigate any possible issues. 

Example: for this particular case, would it have been possible to split out Android Auto Vs CarPlay changes? we'd have different folks reviewing each platform, so that would also make it easier for them to track.

**Comment by ArturoSalazarB16** [comment]
Hi @ShirwaM! If you get some bandwidth, can you please take a look at the iOS portion of the PR? (And also if you have any feedback around Android Auto implementation that would be amazing).

thanks!
Arturo

File: android/src/main/java/com/google/android/react/navsdk/NavAutoModule.java
Comment (ArturoSalazarB16) [inline android/src/main/java/com/google/android/react/navsdk/NavAutoModule.java:43]: Nit: Is there a way to consolidate the methods here with the other NavModule?

**Review by ArturoSalazarB16 — COMMENTED** [review]
Hi! Sharing some initial comments, will take another pass today

File: src/shared/useModuleListeners.ts
Comment (ArturoSalazarB16) [inline src/shared/useModuleListeners.ts:49]: For some of these new files, let's please start building our unit test coverage.

File: src/auto/useNavigationAuto.ts
Comment (ArturoSalazarB16) [inline src/auto/useNavigationAuto.ts:70]: Nit: Is this "Android only" surfacing to consumers?

**Review by ArturoSalazarB16 — APPROVED** [review]
LGTM - to unblock the change. But please let's await for @ShirwaM to provide feedback before merging.

File: android/src/main/java/com/google/android/react/navsdk/AndroidAutoBaseScreen.java
Comment (illuminati1911) [reply android/src/main/java/com/google/android/react/navsdk/AndroidAutoBaseScreen.java:47]: @jokerttu Is this something that will appear on the display or just internal name that won't affect anything?

File: ANDROIDAUTO.md
Comment (illuminati1911) [reply ANDROIDAUTO.md]: Done.

File: ANDROIDAUTO.md
Comment (illuminati1911) [reply ANDROIDAUTO.md]: Fixed.

File: ANDROIDAUTO.md
Comment (illuminati1911) [reply ANDROIDAUTO.md]: Fixed.

File: android/src/main/java/com/google/android/react/navsdk/AndroidAutoBaseScreen.java
Comment (illuminati1911) [reply android/src/main/java/com/google/android/react/navsdk/AndroidAutoBaseScreen.java]: Changed the name to something more generic for now.

File: android/src/main/java/com/google/android/react/navsdk/AndroidAutoBaseScreen.java
Comment (illuminati1911) [reply android/src/main/java/com/google/android/react/navsdk/AndroidAutoBaseScreen.java:54]: Done.

File: android/src/main/java/com/google/android/react/navsdk/AndroidAutoBaseScreen.java
Comment (illuminati1911) [reply android/src/main/java/com/google/android/react/navsdk/AndroidAutoBaseScreen.java]: This is to invalidate layout because of the conditional template rendering. It's actually not supposed to be here. I moved it to the example app with comment.

File: android/src/main/java/com/google/android/react/navsdk/AndroidAutoBaseScreen.java
Comment (jokerttu) [reply android/src/main/java/com/google/android/react/navsdk/AndroidAutoBaseScreen.java:204]: This was not needed anymore. removed

File: src/auto/useNavigationAuto.ts
Comment (jokerttu) [reply src/auto/useNavigationAuto.ts:70]: The documentation for appfacing setMapToolbarEnabled method had misleading documentation. Even when it is not part of thie PR, i updated the documentation as well, including the information that this is Android only feature

File: android/src/main/java/com/google/android/react/navsdk/NavAutoModule.java
Comment (jokerttu) [reply android/src/main/java/com/google/android/react/navsdk/NavAutoModule.java:43]: Not an easy way as NavAutoModule is implementing same functionality as NavViewModule. And as normal nav views can be controlled directly with RN view manager, the functionality differs.

NavModule handles the navigation; and this functionality is separate from view controls.

File: example/android/app/src/main/java/com/sampleapp/SampleAndroidAutoScreen.java
Comment (ShirwaM) [inline example/android/app/src/main/java/com/sampleapp/SampleAndroidAutoScreen.java:84]: Does this lead to the right Maneuver? The `int` values for the Maneuvers in StepInfo don't correspond 1:1 with the Android Auto ones so we'll need a converter here. 

See https://developers.google.com/maps/documentation/navigation/android-sdk/android-auto?section=surface-callbacks#expandable-1

File: example/android/app/src/main/java/com/sampleapp/SampleAndroidAutoScreen.java
Comment (ShirwaM) [inline example/android/app/src/main/java/com/sampleapp/SampleAndroidAutoScreen.java:63]: nit: remove everything after `For more information see the....` portion as the section it's referring to is not in the code here.

File: example/android/app/src/main/java/com/sampleapp/SampleAndroidAutoScreen.java
Comment (ShirwaM) [inline example/android/app/src/main/java/com/sampleapp/SampleAndroidAutoScreen.java:84]: It would be good to show how adding an icon works too. The bitmaps in StepInfo are compatible here so no need to convert and you can do something like: 

```
  IconCompat maneuverIcon =
               IconCompat.createWithBitmap(stepInfo.getManeuverBitmap());
  CarIcon maneuverCarIcon = new CarIcon.Builder(maneuverIcon).build();
  maneuverBuilder.setIcon(maneuverCarIcon);
```

File: example/android/app/src/main/java/com/sampleapp/SampleAndroidAutoScreen.java
Comment (ShirwaM) [inline example/android/app/src/main/java/com/sampleapp/SampleAndroidAutoScreen.java:54]: nit: I don't think this comment adds much value here since line 50 above is self documenting.

**Review by ShirwaM — CHANGES_REQUESTED** [review]
LGTM pending a few minor change requests. Thank you!

File: example/android/app/src/main/java/com/sampleapp/SampleAndroidAutoScreen.java
Comment (illuminati1911) [reply example/android/app/src/main/java/com/sampleapp/SampleAndroidAutoScreen.java]: Fixed.

File: example/android/app/src/main/java/com/sampleapp/SampleAndroidAutoScreen.java
Comment (illuminati1911) [reply example/android/app/src/main/java/com/sampleapp/SampleAndroidAutoScreen.java]: Fixed.

File: example/android/app/src/main/java/com/sampleapp/SampleAndroidAutoScreen.java
Comment (illuminati1911) [reply example/android/app/src/main/java/com/sampleapp/SampleAndroidAutoScreen.java]: Done.

File: example/android/app/src/main/java/com/sampleapp/SampleAndroidAutoScreen.java
Comment (illuminati1911) [reply example/android/app/src/main/java/com/sampleapp/SampleAndroidAutoScreen.java]: Removed.

**Review by zioncalvo — APPROVED** [review]
LGTM for iOS specific changes

**Pull request merged**
